### PR TITLE
fix: discover search lag

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes RNBW-3181

## What changed (plus any additional context for devs)

if you search for an ENS name, clear the query and search again multiple instances of the same row repeat
for some reason ExchangeAssetSectionList doesn't update correctly on data change, so I added a key depending on search result to force updates

## PoW (screenshots / screen recordings)

before
https://recordit.co/GsY0MC2n7D

after
https://recordit.co/mJjwwSvL8M

## Final checklist

- [ ] Assigned individual reviewers?
- [ ] Added labels?
- [ ] Added e2e tests? if not please specify why
- [ ] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

File: src/components/discover-sheet/DiscoverSearch.js [hunk 1]
```diff
@@ -58,6 +58,14 @@ export default function DiscoverSearch() {
     ensResults,
   ]);
 
+  const currencyListDataLength = useMemo(
+    () =>
+      uniswapCurrencyList?.[0]?.data?.length ||
+      0 + ensResults?.[0]?.data?.length ||
+      0,
```
Comment (skylarbarrera) [inline src/components/discover-sheet/DiscoverSearch.js:65]: nit: can we do a string so theres no possibility of the length being the same? something like `${uniswapCurrencyList?.[0]?.data?.length}_${ensResults?.[0]?.data?.length}`

**Review by skylarbarrera — COMMENTED** [review]
👌

**Review by skylarbarrera — APPROVED** [review]

File: src/components/discover-sheet/DiscoverSearch.js (hunk 1 above)
Comment (osdnk) [reply src/components/discover-sheet/DiscoverSearch.js:65]: yes, I don't want to be a pain in the ass, but this is not 100% unique

**Review by osdnk — CHANGES_REQUESTED** [review]

File: src/components/discover-sheet/DiscoverSearch.js (hunk 1 above)
Comment (estebanmino) [reply src/components/discover-sheet/DiscoverSearch.js]: 💯  done!

**Comment by estebanmino** [comment]
@osdnk is that the new rebase gh action?

**Comment by terrysahaidak** [comment]
I think the problem is in `CurrencySelectionList`, it has `magicMemo` but not the `query` as its dependency.

**Comment by terrysahaidak** [comment]
@estebanmino pinging you regarding my last comment :)

**Comment by estebanmino** [comment]
oh sorry! @terrysahaidak   taking a look now

**Comment by estebanmino** [comment]
@terrysahaidak i was still seeing this happening with the `query` dep in magicMemo feel like something is off but i'm not sure about the internals there, added it anyways. 
I changed the key as @osdnk comment to make it unique on search results taking addresses

**Review by terrysahaidak — APPROVED** [review]

**Review by osdnk — APPROVED** [review]

**Comment by estebanmino** [comment]
waiting for profiles to get merged

**Pull request merged**